### PR TITLE
add pytest and port over old tests

### PR DESCRIPTION
**Pull request by abrown117**
This will make running tests in ci a lot simpler.

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -1,32 +1,27 @@
...
+from PyQt5.QtCore import Qt
 from PyQt5.QtWidgets import QApplication
 from PyQt5.QtTest import QTest
 
 from uaclient.mainwindow import Window
 
 
-class TestClient(unittest.TestCase):
-    def setUp(self):
+class TestClient:
```
Comment (kyrofa) [inline tests/test_client.py]: Is there a benefit to having a class, here?

File: requirements.txt [hunk 2]
```diff
@@ -3,14 +3,22 @@ aiosqlite==0.20.0
 asyncua==1.1.5
 cffi==1.17.1
 cryptography==43.0.1
+exceptiongroup==1.2.2
```
Comment (kyrofa) [inline requirements.txt]: Let's please keep `requirements.txt` to mean "run-time requirements." Make a new `requirements-dev.txt` for these things, and update the workflow to install it as well.

If these are all dependencies from pytest, remove them entirely and only have pytest in the requirements-dev.txt, please.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[tool.pytest.ini_options]
+pythonpath = [".", "uaclient"]
```
Comment (kyrofa) [inline pyproject.toml]: I suspect this isn't actually necessary if you have the package installed before you run tests, can you confirm?

**Review by kyrofa — CHANGES_REQUESTED** [review]

File: tests/test_client.py (hunk 1 above)
Comment (abrown117) [reply tests/test_client.py]: Other than matching the original tests not really. It should work the same either way, with some minor refactoring.

Comment (kyrofa) [reply tests/test_client.py]: Scrap the format of the original tests. Focus on establishing a pattern to follow for the rest of the codebase.

File: requirements.txt [hunk 4]
```diff
@@ -4,6 +4,7 @@ asyncua==1.1.5
 cffi==1.17.1
 cryptography==43.0.1
 opcua-widgets==0.6.1
+packaging==24.2
```
Comment (kyrofa) [inline requirements.txt]: Is this a runtime requirement?

**Review by kyrofa — APPROVED** [review]
This looks great, thank you!

**Pull request merged**